src/main/java/seedu/address/model/delivery/Delivery.java

```diff
@@ -89,14 +123,16 @@ public boolean equals(Object other) {
         if (!(other instanceof Delivery)) {
             return false;
         }
-
         Delivery otherDelivery = (Delivery) other;
+        boolean isSameSender = this.sender == null || otherDelivery.sender == null
```

Comment:
Instead of having null as the default sender should we have some static final sender that indicates that this delivery does not have sender associated with it. Furthermore, is this check even necessary since we do the requireAllNonNull check when constructing the delivery.

Reply:
Oh the check was for the test cases as the assertEqual method will not work if the sender is null even those both deliveries are equal.

I think having a default sender would work as well instead of having null.